shared-bindings/memorymonitor/AllocationSize.c

```diff
@@ -0,0 +1,183 @@
+    memorymonitor_allocationsize_obj_t *self = m_new_obj(memorymonitor_allocationsize_obj_t);
+    self->base.type = &memorymonitor_allocationsize_type;
+
+    common_hal_memorymonitor_allocationsize_construct(self);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def __enter__(self, ) -> Any:
+//|         """Clears counts and resumes tracking."""
+//|         ...
+//|
+STATIC mp_obj_t memorymonitor_allocationsize_obj___enter__(mp_obj_t self_in) {
+    common_hal_memorymonitor_allocationsize_clear(self_in);
+    common_hal_memorymonitor_allocationsize_resume(self_in);
+    return self_in;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(memorymonitor_allocationsize___enter___obj, memorymonitor_allocationsize_obj___enter__);
+
+//|     def __exit__(self, ) -> Any:
+//|         """Automatically pauses allocation tracking when exiting a context. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+//|
+STATIC mp_obj_t memorymonitor_allocationsize_obj___exit__(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+    common_hal_memorymonitor_allocationsize_pause(args[0]);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(memorymonitor_allocationsize___exit___obj, 4, 4, memorymonitor_allocationsize_obj___exit__);
+
+//|     bytes_per_block: int = ...
+//|     """Number of bytes per block"""
+//|
+STATIC mp_obj_t memorymonitor_allocationsize_obj_get_bytes_per_block(mp_obj_t self_in) {
+    memorymonitor_allocationsize_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    return MP_OBJ_NEW_SMALL_INT(common_hal_memorymonitor_allocationsize_get_bytes_per_block(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(memorymonitor_allocationsize_get_bytes_per_block_obj, memorymonitor_allocationsize_obj_get_bytes_per_block);
+
+const mp_obj_property_t memorymonitor_allocationsize_bytes_per_block_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&memorymonitor_allocationsize_get_bytes_per_block_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|     def __len__(self, ) -> Any:
+//|         """Returns the number of allocation buckets.
+//|
+//|         This allows you to::
+//|
+//|           mm = memorymonitor.AllocationSize()
+//|           print(len(mm))"""
+//|         ...
+//|
+STATIC mp_obj_t memorymonitor_allocationsize_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
+    memorymonitor_allocationsize_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    uint16_t len = common_hal_memorymonitor_allocationsize_get_len(self);
+    switch (op) {
+        case MP_UNARY_OP_BOOL: return mp_obj_new_bool(len != 0);
+        case MP_UNARY_OP_LEN: return MP_OBJ_NEW_SMALL_INT(len);
+        default: return MP_OBJ_NULL; // op not supported
+    }
+}
+
+//|     def __getitem__(self, index: Any) -> Any:
+//|         """Returns the allocation count for the given bucket.
+//|
+//|         This allows you to::
```

Comment:
double colon

Reply:
This is an RST thing: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks